hicommonwealth/commonwealth
# Autogenerate stakeholder group

## Link to Issue
Closes: #6843
Closes: #7001

## Description of Changes
- Adds script to generate stakeholder group for existing staked communities
- Implements policy to autogenerate stakeholder group when a stake is enabled on a community
- Splits out `searchProfiles` into 2 separate routes: the original route + a new route specifically for "community members"
  - Main reason is due to performance– the original route is more performant and used for search bar while the new route is for the members page and includes groups, staked balances, etc.
 - Adds "Stake" column to members page UI

## Test Plan
- CA Testing
  - Run migration
  - Create a staked community
  - Buy some stake
  - Go to members page, confirm that your profile shows with `Stakeholder` label and stake amount visible
  - Go to groups page, confirm that `Stakeholder` group exists
  - In DB, delete the stakeholder group, then from `packages/scripts`, run `yarn generate-stakeholder-groups`
    - Confirm that the stakeholder group is recreated

## Deployment Plan
- From the `packages/scripts` directory, run `yarn add ts-node tsconfig-paths && yarn generate-stakeholder-groups` to create stakeholder groups for existing staked communities

## Other Considerations
- The chosen implementation is the simplest, but comes at the cost of performance. It must fetch *all* members of the community and their stake balances even though the request is paginated. This is necessary because stake balances are fetched from onchain, and must be sorted in memory.
